--- cmodel_rtl_reference/nonlinear_cmodel_reference.py
+++ cmodel_rtl_reference/nonlinear_cmodel_reference.py
@@ -52,6 +52,9 @@
     q = x_int // x0_int
     r = x_int - x0_int * q
 
+    if not np.issubdtype(r.dtype, np.integer):
+        print("CRASH IN EXP! r type:", r.dtype, "q type:", q.dtype, "x0_int type:", type(x0_int), "x_int type:", x_int.dtype)
+
     # 基礎乘數
     exp_int = (r >> 1) - x0_int
     shift = n - q
